controller/lib/core/sprintf.cpp

```diff
@@ -439,9 +439,9 @@ static int FormatLong(FieldInfo *info, long long val, char *str, int max) {
 static int FormatFloat(FieldInfo *info, float val, char *str, int max) {
   // Check for outliers
   const char *bad = 0;
-  if (isnanf(val))
+  if (isnan(val))
     bad = (val < 0) ? "-nan" : "nan";
-  if (isinff(val))
+  if (isinf(val))
```

Comment:
The only problem is that this will cause the input argument to be promoted to a double every time we use one.  Maybe this doesn't matter too much since this is only for debugging, but the use of doubles in this environment makes me winch.  We already have to convert to double and back once because variable args don't support floats.
An alternative would be to add a conditionally compiled section to the top of the file that replaces isinff with isinf on your machine but leaves it as is for the STM32.  That would avoid the floating point conversion on the production machine, but still allow you to use it on your Linux box.
Anyway, it's just debug code so I'll leave it up to you whether you do that or not.

Reply:
> The only problem is that this will cause the input argument to be promoted to a double every time we use one. 

Hm, https://en.cppreference.com/w/c/numeric/math/isnan suggests that this function does not necessarily force an upcast to double.  (It also doesn't say it *doesn't* force an upcast...)

std::isnan has an overload for float and definitely doesn't cause an upcast to double, https://en.cppreference.com/w/cpp/numeric/math/isnan

I checked the codegen with our flags and `::isnan` and `std::isnan` generate the same code.  I'm not familiar enough with the ARM ISA to say whether there's an upcast going on.

https://gcc.godbolt.org/z/AA_6P-

If you want I can switch it to std::isnan?

Reply:
(Gonna push this so the build is not broken, but I'm happy to go back and add `std::` [or do some other change])

Reply:
Which ever you prefer.  It's just debug code after all.  If I later find that it does upcast to double on the STM32 I may add some conditional compile stuff to get around that.  Or I might not, it would be very low priority for me and I've got a lot of more important/interesting stuff to work on.